File: src/Azure/Mgmt/Microsoft.Compute/2016_06_30/VirtualMachines/Models/VirtualMachine.cs
using Azure.Mgmt.Models;

namespace Azure.Mgmt.Compute._2016_06_30.Models
{
    public class VirtualMachine : Azure.Mgmt.Compute._2016_01_31.Models.VirtualMachine
    {
        /// <summary>
        /// Gets specifies the VM unique ID which is a 128-bits identifier that
        /// is encoded and stored in all Azure IaaS VMs SMBIOS and can be read
        /// using platform BIOS commands.
        /// </summary>
        public string VmId { get; protected set; }
    }
}

File: ruby/src/azure/mgmt/microsoft_compute/2016_01_31/virtual_machines/models/virtual_machine.rb
require_relative "../../../models/virtual_machine"

module Azure
  module Mgmt
    module Compute
      module Api_2016_01_31
        module Models
          class VirtualMachine < Azure::Mgmt::Compute::Models::VirtualMachine
            attr_accessor :vm_id

            def initialize
              super
              self.vm_id = "vm_id inside of Api_2016_01_31"
            end
          end
        end
      end
    end
  end
end